File: modelsqlite/src/main/java/com/zpf/modelsqlite/constant/SQLiteConfig.java
package com.zpf.modelsqlite.constant;

import android.app.Application;
import android.content.pm.ApplicationInfo;

public class SQLiteConfig {
    private SQLiteConfig() {
    }
    public static final int DB_CACHE_VERSION = 1;
    public static final String DB_USER_CACHE = "userCache.db";
    public static final String TB_CACHE = "userCache";
    public static final String PRIMARY_KEY = "_id"; //主键
    public static final String FIRST_KEY_WORD = "keyword"; //表名
    public static final String OTHER = "other"; //预留列
    public static final String TIME_CREATE = "create_time"; //数据插入时间--long类型
    public static final String TIME_UPDATE = "update_time"; //数据更新时间--long类型
    public static final String COLUMN_NAME = "column";

    public static final int TYPE_SHORT = 1000;
    public static final int TYPE_DOUBLE = 2000;
    public static final int TYPE_LONG = 3000;
    public static final int TYPE_FLOAT = 4000;
    public static final int TYPE_BOOLEAN = 5000;
    public static final int TYPE_INT = 6000;
    public static final int TYPE_STRING = 7000;

    public static final int DEF_NUMBER = 0;
    public static final boolean DEF_BOOLEAN = false;
    public static final String DEF_STRING = "";

    public static final String COMMA = " , ";
    public static final String ORDER = " ORDER BY ";
    public static final String GROUP = " GROUP BY ";
    public static final String HAVING = " HAVING ";
    public static final String ASC = " ASC";
    public static final String DESC = " DESC";
    public static final String LIMIT = " LIMIT ";
    public static final String AND = " AND ";
    public static final String OR = " OR ";

    public static final String WHERE = " WHERE " + FIRST_KEY_WORD + " = ";
    public static final String VALUES = ")VALUES(";
    public static final String SQL_SELECT = "SELECT * FROM " + TB_CACHE + WHERE;
    public static final String SQL_UPDATE = "UPDATE " + TB_CACHE + " SET ";
    public static final String SQL_INSERT = "INSERT INTO " + TB_CACHE + " (";
    public static final String SQL_DELETE = "DELETE FROM " + TB_CACHE + WHERE;

    private static Application appContext;
    public static String DB_PATH = null;

    public static void initConfig(Application app, String dbPath) {
        if (app != null) {
            appContext = app;
        }
        DB_PATH = dbPath;
    }

    public static Application getAppContext() {
        return appContext;
    }

}

File: modelsqlite/src/main/java/com/zpf/modelsqlite/constant/SQLiteConfig.kt
package com.zpf.modelsqlite.constant

import android.app.Application

/**
 * 数据可辅助常量
 * Created by ZPF on 2018/1/24.
 */
object SQLiteConfig {
    const val DB_CACHE_VERSION = 1
    const val DB_USER_CACHE = "userCache.db"
    const val TB_CACHE = "userCache"
    const val PRIMARY_KEY = "_id" //主键
    const val FIRST_KEY_WORD = "keyword" //表名
    const val OTHER = "other" //预留列
    const val TIME_CREATE = "create_time" //数据插入时间--long类型
    const val TIME_UPDATE = "update_time" //数据更新时间--long类型
    const val COLUMN_NAME = "column"

    const val TYPE_SHORT = 1000
    const val TYPE_DOUBLE = 2000
    const val TYPE_LONG = 3000
    const val TYPE_FLOAT = 4000
    const val TYPE_BOOLEAN = 5000
    const val TYPE_INT = 6000
    const val TYPE_STRING = 7000

    const val DEF_NUMBER = 0
    const val DEF_BOOLEAN = false
    const val DEF_STRING = ""

    const val COMMA = " , "
    const val ORDER = " ORDER BY "
    const val GROUP = " GROUP BY "
    const val HAVING = " HAVING "
    const val ASC = " ASC"
    const val DESC = " DESC"
    const val LIMIT = " LIMIT "
    const val AND = " AND "
    const val OR = " OR "

    const val WHERE = " WHERE $FIRST_KEY_WORD = "
    const val VALUES = ")VALUES("
    const val SQL_SELECT = "SELECT * FROM $TB_CACHE$WHERE"
    const val SQL_UPDATE = "UPDATE $TB_CACHE SET "
    const val SQL_INSERT = "INSERT INTO $TB_CACHE ("
    const val SQL_DELETE = "DELETE FROM $TB_CACHE$WHERE"

    internal var DB_PATH: String? = null
    internal var appContext: Application? = null
    internal var LOG_TAG = "ModelSQLite"
    internal var DEBUG = false
}